### ISSUE TITLE
bug(H006): not possible to ignore if use HTML comment syntax or ignoring comments inside a tag

### ISSUE THREAD

**Issue by Kristinita** [Riverside-Healthcare/djLint#344]
**#333** — possibly related issue

### 1. Summary

I can’t ignore H006 error **if**:

1. I use `<!-- HTML comments syntax -->`.
1. **Or if** my ignoring comments inside a tag.

#### 2.1. Files to check

Content of my `KiraH006` folder:

1. `KiraDJLint__Ignore-specific-by-HTML-comments--H006.html`:

	```jinja
	<!-- djlint:off T027,H017,H006 -->

	{% if "<img alt=\"" in kira_article %}
		<script src="KiraJS.js" defer></script>
	{% endif %}

	<!-- djlint:on -->

	```

1. `KiraDJLint__Ignore-all-by-HTML-comments--H006.html`:

	```jinja
	<!-- djlint:off -->

	{% if "<img alt=\"" in kira_article %}
		<script src="KiraJS.js" defer></script>
	{% endif %}

	<!-- djlint:on -->

	```

1. `KiraDJLint__Ignore-H006-inside-div--H006.html`:

	```jinja
	<div>

			{# djlint:off H006 #}

			<img src="{{ kira_variable }}.webp" alt="Kira Goddess!" />

			{# djlint:on #}

	</div>

	```

1. `KiraDJLint__Ignore-all-inside-div--H006.html`:

	```jinja
	<div>

			{# djlint:off #}

			<img src="{{ kira_variable }}.webp" alt="Kira Goddess!" />

			{# djlint:on #}

	</div>

	```

#### 2.2. Desired behavior

No errors

#### 2.3. Current behavior

```text
D:\SashaDebugging\KiraDJLintDebuging>djlint --profile=jinja --lint KiraH006

Linting 4/4 files ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 00:00


SashaDebugging\KiraDJLintDebuging\KiraH006\KiraDJLint__Ignore-H006-inside-div--H006.html
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── H006 5:2 Img tag should have height and width attributes. <img src="{{ kira_va

SashaDebugging\KiraDJLintDebuging\KiraH006\KiraDJLint__Ignore-all-by-HTML-comments--H006.html
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── H006 3:7 Img tag should have height and width attributes. <img alt=\"" in kira

SashaDebugging\KiraDJLintDebuging\KiraH006\KiraDJLint__Ignore-all-inside-div--H006.html
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── H006 5:2 Img tag should have height and width attributes. <img src="{{ kira_va

SashaDebugging\KiraDJLintDebuging\KiraH006\KiraDJLint__Ignore-specific-by-HTML-comments--H006.html
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── H006 3:7 Img tag should have height and width attributes. <img alt=\"" in kira

Linted 4 files, found 4 errors.
```

### 3. Environment

1. Microsoft Windows [Version 10.0.19041.1415]
1. Python 3.10.6
1. djlint, version 1.11.0
1. Template language — Jinja

### 4. Checkbox

1. [x] I’m on the [**latest version**](https://pypi.org/project/djlint/) of djLint
1. [x] I’ve searched the [**issues**](https://github.com/Riverside-Healthcare/djLint/issues)
1. [x] I’ve read the [**docs**](https://djlint.com)

Thanks.

**Comment by christopherpickering** [comment]
@Kristinita Thanks, I can't get this to reproduce. I tested with Windows 10, Python 3.10, and djlint 1.11.1. I also try with ubuntu + python3.8 + djlint 3.11.1 and osx with py 3.9 + djlint 1.11.1.

@aleprovencio

Can either of you help with more debugging? 

Specifically, I'm interesting in what is going on in `lint.py` on lines 124 when the `inside_ignored_rule` is called.

On line 122 can you add `print(inside_ignored_rule(config, html, match,rule["name"]))` ?

In `helpers.py` on line 143 can you add `print(rule_regex)` and try to find which regex is not matching but should be? From the examples in this issue, it sounds like none of them are matching..

Thanks!

**Comment by Kristinita** [comment]
**Status: Partially fixed** :moon:

### 1. Summary

@christopherpickering , for djLint 1.11.1 I can’t reproduce problems in examples 1, 2 and 4. Thanks!

But I still get H006 error for the example 3 on my machine.

### 2. Circle CI Ubuntu testing

I reproduce the problem on the Circle CI Ubuntu.

#### 2.1. Configuration

You can see this configuration on the [**KiraDJLintH006 branch**](https://github.com/Kristinita/SashaCircle/tree/KiraDJLintH006) of my demo/debugging repository.

+ `config.yml`

	```yaml
	version: 2.1

	jobs:
	  build:
	    machine:
	      image: ubuntu-2204:current
	    steps:
	    - checkout
	    - run: pyenv global 3.10.5
	    - run: pip install djlint
	    - run: djlint --profile=jinja KiraDJLint__Ignore-H006-inside-div--H006.html

	```

#### 2.2. Build

[**Build**](https://app.circleci.com/pipelines/github/Kristinita/SashaCircle/1/workflows/372cdcb8-2583-4eca-b241-4963d628f48d/jobs/13), [**result of executing djLint command**](https://app.circleci.com/pipelines/github/Kristinita/SashaCircle/1/workflows/372cdcb8-2583-4eca-b241-4963d628f48d/jobs/13/parallel-runs/0/steps/0-104):

```text
djlint --profile=jinja KiraDJLint__Ignore-H006-inside-div--H006.html

100% 1/1 [00:00<?, ?it/s]


KiraDJLint__Ignore-H006-inside-div--H006.html

H006 5:2 Img tag should have height and width attributes. <img src="{{ kira_va

Linted 1 file, found 1 error.
```

#### 2.3. Environment

1. Ubuntu 22.04 (Jammy Jellyfish)
1. Python 3.10.5
1. djLint 1.1.1

Thanks.

**Comment by christopherpickering** [comment]
Thanks! I was able to reproduce it in ubuntu docker + py 3.10.. so I switched back on the Mac and tested out w/ py.3.10 there and was able to reproduce it as well.. Must be something different in how 3.10 works. I will check it out!

**Comment by christopherpickering** [comment]
But on more testing, I'm able to reproduce in py 3.9 now :D Something must have changed when I copy/paste yesterday.

**Comment by Kristinita** [comment]
**Status: FIXED** :heavy_check_mark: for me for djLint 1.12.0. Also, [**fixed**](https://app.circleci.com/pipelines/github/Kristinita/SashaCircle/1/workflows/298437b5-2646-46ea-9804-11055c271208/jobs/14) on Circle CI Ubuntu.

Thanks.
